packages/common-ts/src/base-service/base-service-v2.ts

```diff
@@ -0,0 +1,130 @@
+/* Imports: External */
+import Config from 'bcfg'
+import * as dotenv from 'dotenv'
+import { Command, Option } from 'commander'
+import { ValidatorSpec, cleanEnv } from 'envalid'
+import { sleep } from '@eth-optimism/core-utils'
+
+/* Imports: Internal */
+import { Logger } from '../common/logger'
+
+export type Options = {
+  [key: string]: any
+}
+
+export type OptionsSpec<TOptions extends Options> = {
+  [P in keyof TOptions]: ValidatorSpec<TOptions[P]>
+}
+
+export abstract class BaseServiceV2<TOptions extends Options, TServiceState> {
+  protected loop: boolean
+  protected loopIntervalMs: number
+  protected logger: Logger
+  protected state: TServiceState
+  protected readonly options: TOptions
+
+  /**
+   * @param params Options for the construction of the service.
+   * @param params.name Name for the service. This name will determine the prefix used for logging,
+   * metrics, and loading environment variables.
+   * @param params.optionsSpec Settings for input options. You must specify at least a
+   * description for each option.
+   * @param params.options Options to pass to the service.
+   * @param params.loops Whether or not the service should loop. Defaults to true.
+   * @param params.loopIntervalMs Loop interval in milliseconds. Defaults to zero.
+   */
```

Comment:
Note to self: I'd like a nice way of handling "service containers" -- groups of services that run together. Not sure of the best way to represent this. I'm not particularly happy with the current thing of running a service that runs other services.